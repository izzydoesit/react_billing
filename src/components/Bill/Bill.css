li.bill-item {
  padding: 1%;
  margin: 1%;
  border: 2px solid #949598;
  width: 40%;
}