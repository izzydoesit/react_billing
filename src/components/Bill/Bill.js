// @flow

import React from 'react';
import './Bill.css';

type Props = {
  id: string,
  status: string,
  amount: string,
  dueDate: string,
}

const Bill = (props: Props) => {

  const {
    id,
    status,
    amount,
    dueDate } = props;

  return <li className="bill-item">
            <span className="bill-id">{id}: </span> 
            <span className="bill-amount">{amount}, </span> 
            <span className="bill-status">{status}, </span> 
            <span className="bill-due-date">{dueDate}</span>
          </li>
}

export default Bill;
