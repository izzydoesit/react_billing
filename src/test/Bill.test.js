// @flow

import React from 'react';
import Bill from '../components/Bill/Bill';
import { formatCentsToDollars } from 'clark-utils';
import { mount, configure } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<Bill />', () => {
  let props, billText;
  let mountedBill;
  const bill = () => {
    if (!mountedBill) {
      mountedBill = mount(<Bill {...props} />);
    }
    return mountedBill;
  };

  beforeEach(() => {
    props = {
      id: undefined,
      status: undefined,
      amount: undefined,
      dueDate: undefined,
    }
    mountedBill = undefined;
  });

  it('renders correctly', () => {
    expect(mountToJson(bill())).toMatchSnapshot();
  });

  it('renders an <li> element', () => {
    expect(bill().find('li.bill-item').length).toEqual(1);
  });

  describe('when id is defined', () => {
    beforeEach(() => {
      props.id = 'myKey';
      billText = bill().text();
    });

    it('renders an id', () => {
      expect(billText).toContain(props.id);
    });
  });
 
  describe('when status is defined', () => {
    beforeEach(() => {
      props.status = 'some status';
      billText = bill().text();
    });

    it('renders a status', () => {
      expect(billText).toContain(props.status);
    });
  });
 
  describe('when amount is defined', () => {
    beforeEach(() => {
      props.amount = formatCentsToDollars(12345);
      billText = bill().text();
    });

    it('renders an amount', () => {
      expect(billText).toContain(props.amount);
    });
  });
 
  describe('when dueDate is defined', () => {
    beforeEach(() => {
      props.dueDate = '2000-01-01';
      billText = bill().text();
    });

    it('renders a due date', () => {
      expect(billText).toContain(props.dueDate);
    });
  });
});
