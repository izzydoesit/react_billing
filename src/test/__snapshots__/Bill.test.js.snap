// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bill /> renders correctly 1`] = `
<Bill>
  <li
    className="bill-item"
  >
    <span
      className="bill-id"
    >
      : 
    </span>
    <span
      className="bill-amount"
    >
      , 
    </span>
    <span
      className="bill-status"
    >
      , 
    </span>
    <span
      className="bill-due-date"
    />
  </li>
</Bill>
`;
